test/unit-tests/common/test_image_star.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+"""
+    :copyright: Copyright 2017-2019 by the contributors (see AUTHORS file).
+    :license: BSD-2-Clause, see LICENSE for details.
+"""
+
+from sphinxcontrib.confluencebuilder.state import ConfluenceState
+from sphinxcontrib_confluencebuilder_util import ConfluenceTestUtil as _
+from sphinxcontrib.confluencebuilder.builder import ConfluenceBuilder
+import os
+import re
+import unittest
+import tempfile
+import shutil
+
+
+class TestImageStar(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.extension_pattern = '@REPLACE_EXTENSION@'
+        cls.test_dir = os.path.dirname(os.path.realpath(__file__))
+        cls.image_file = os.path.join(cls.test_dir, 'assets', 'image01.png')
+
+    def build_document_given_dataset(self, dataset):
+        config = _.prepareConfiguration()
+        doc_dir, doctree_dir = _.prepareDirectories('image-star')
+        _.buildSphinx(dataset, doc_dir, doctree_dir, config)
+        return doc_dir
+
+    def write_expected_file(self, pattern_file, outfile, actual_extension):
+        ''' writes the content of pattern_file into outfile replacing
+        the image pattern by the actual extension
+        '''
+        with open(pattern_file) as inp, open(outfile, 'w') as out:
+            content = ''.join(inp.readlines())
+            out.write(content.replace(self.extension_pattern, actual_extension))
+
+    def test_all_supported_extensions(self):
```

Comment:
A suggestion, if I may, for this test. I looks like you are following the convention found in other tests, were you have a data set, generate results and check on an pre-built expected file for verification; however, you are modifying for each image check the expected file and generating new documentation outputs for each type defined in `supported_image_types`. Why not instead just drop `expected-image-star` completely and generate the entire "expected" document and dummy image types in another temporary directory? This could allow you to have just a single dynamic "expected" document which you only need to perform a single Sphinx build event on (rather than building `len(StandaloneHTMLBuilder.supported_image_types)` times).

Reply:
Eventually, I realized that to test an image type, you need an actual image with that image type. It seems the Builder discovers the image candidates according to the actual file type, not only the extension. For every candidate, there is a path/uri assigned. I was testing always the same candidate type (png) with different path assigned.